src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py

```diff
@@ -0,0 +1,248 @@
+            measurement_document=measurement_documents,
+            plate_well_count=TQuantityValueNumber(value=0),
+        )
+        return DPCRDocumentItem(
+            measurement_aggregate_document=measurement_aggregate_document
+        )
+
+    def _get_device_system_document(self) -> DeviceSystemDocument:
+        device_system_document = DeviceSystemDocument(
+            device_identifier=DEVICE_IDENTIFIER,
+            brand_name=BRAND_NAME,
+            product_manufacturer=PRODUCT_MANUFACTURER,
+        )
+        return device_system_document
+
+    def _get_data_system_document(self, file_name: str) -> DataSystemDocument:
+        data_system_document = DataSystemDocument(
+            file_name=file_name,
+            software_name=SOFTWARE_NAME,
+            ASM_converter_name=ASM_CONVERTER_NAME,
+            ASM_converter_version=ASM_CONVERTER_VERSION,
+        )
+        return data_system_document
+
+    def _get_measurement_document(
+        self,
+        well_item: pd.Series,
+        sample_document: SampleDocument,
+        device_control_aggregate_document: DeviceControlAggregateDocument,
+        processed_data_aggregate_document: ProcessedDataAggregateDocument,
+    ) -> MeasurementDocumentItem:
+        # Assign a random measurement id
+        measurement_id = random_uuid_str()
+        # There is no measurement time in the file, so assign to unix epoch
+        measurement_time = EPOCH
+        target_dna_description = self._get_value_by_col_name(
+            well_item, TARGET_COLUMN_NAME, required=True
+        )
+        total_partition_count = TQuantityValueNumber(
+            self._get_value_by_col_name(
+                well_item, PARTITIONS_COLUMN_NAME, required=True
+            )
+        )
+        return MeasurementDocumentItem(
+            measurement_identifier=measurement_id,
+            measurement_time=measurement_time,
+            target_DNA_description=target_dna_description,
+            total_partition_count=total_partition_count,
+            sample_document=sample_document,
+            device_control_aggregate_document=device_control_aggregate_document,
+            processed_data_aggregate_document=processed_data_aggregate_document,
+        )
+
+    def _get_sample_document(self, well_item: pd.Series) -> SampleDocument:
+        sample_identifier = str(
+            self._get_value_by_col_name(
+                well_item, SAMPLE_IDENTIFIER_COLUMN_NAME, required=True
+            )
+        )
+        sample_document = SampleDocument(sample_identifier=sample_identifier)
+
+        sample_role_type = self._get_value_by_col_name(
+            well_item, SAMPLE_TYPE_COLUMN_NAME, required=False
+        )
+        # TODO: When the sample role type model is updated in this repo, we should update this
+        # Map sample role types to valid sample role types from ASM
+        if sample_role_type is not None:
+            sample_role_type = SAMPLE_ROLE_TYPE_MAPPING[sample_role_type]
```

Comment:
Error handle here if `sample_role_type` is not in mapping (raise `AllotropeConversionError`)

Reply:
This is updated!